sklearn/ensemble/_hist_gradient_boosting/histogram.pyx

```diff
@@ -153,11 +166,18 @@ cdef class HistogramBuilder:
                         ordered_gradients[i] = gradients[sample_indices[i]]
                         ordered_hessians[i] = hessians[sample_indices[i]]
 
-            for feature_idx in prange(n_features, schedule='static',
-                                      num_threads=n_threads):
-                # Compute histogram of each feature
+            # Compute histogram of each feature
+            for f_idx in prange(
+                n_allowed_features, schedule='static', num_threads=n_threads
+            ):
```

Comment:
If the here-above suggested `nogil, parallel`-context is used, this `num_threads` must be removed here and in previous `prange` loops I can't suggest on. 
```suggestion
            for f_idx in prange(n_allowed_features, schedule='static'):
```

Note that this partern

Reply:
In another PR, maybe. See comment above.